File: api/src/main/java/net/blueshell/api/dto/PersonalInfoDTO.java
package net.blueshell.api.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.EqualsAndHashCode;
import net.blueshell.api.base.BaseDTO;
import net.blueshell.api.validation.user.ValidMobilePhoneNumber;

@Data
@EqualsAndHashCode(callSuper = false)
@Schema(name = "PersonalInfo")
public class PersonalInfoDTO extends BaseDTO {
    @NotBlank
    private String discord;

    @NotBlank
    @Email
    private String email;

    @NotBlank
    @ValidMobilePhoneNumber
    private String phoneNumber;
}

File: api/src/main/kotlin/net/blueshell/api/dto/PersonalInfoDTO.kt
package net.blueshell.api.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.persistence.MappedSuperclass
import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank
import net.blueshell.api.dto.base.AuditedAutoIdDTO
import net.blueshell.api.validation.user.ValidMobilePhoneNumber

@Schema(name = "PersonalInfo")
@MappedSuperclass
class PersonalInfoDTO(
    @field:NotBlank
    var discord: String? = null,

    @field:NotBlank
    @field:Email
    var email: String? = null,

    @field:NotBlank
    @field:ValidMobilePhoneNumber
    var phoneNumber: String? = null
) : AuditedAutoIdDTO()
